internal/xds/translator/translator.go

```diff
@@ -54,14 +63,14 @@ func Translate(ir *ir.Xds) (*types.ResourceVersionTable, error) {
 
 		for _, httpRoute := range httpListener.Routes {
 			// 1:1 between IR HTTPRoute and xDS config.route.v3.Route
-			xdsRoute, err := buildXdsRoute(httpRoute)
+			xdsRoute, err := buildXdsRoute(httpRoute, tlsPassthrough)
```

Comment:
this doesnt look right, recommend looking at https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/tls#step-5-test-proxying-https-passthrough to see how its using a `tcp_proxy` filter